source/upgrade.txt

```diff
@@ -0,0 +1,156 @@
+always check for any other differences in options and return values before
+moving it to production.
+
+.. tip::
+
+   To minimize the amount of changes your application may need when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <stable-api-java>`.
+
+.. _java-breaking-changes:
+
+Breaking Changes
+----------------
+
+The breaking changes in this section are categorized by the driver version that
+introduced them.
+
+If you are either upgrading the driver version in your application or plan
+to run multiple applications that use different driver versions, address
+any of the breaking changes between the two versions. For example, if you
+are upgrading the driver from v4.0 to v4.5, address all breaking changes from
+the version after v4.0 including any listed under v4.5.
+
+.. _java-breaking-changes-v4.2:
+
+Version 4.2 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- The ``ObjectId`` class and its ``serialVersionUID`` field were updated
+  to use a new format that minimizes serialization compatibility issues
+  across different versions of the driver.
+
+  Any objects that include an ``ObjectId`` field that were serialized with an
+  older version of the driver are not deserializable by this driver version
+  and later. When the driver encounters an incompatible format, it throws an
+  exception.
+
+.. _java-upgrade-from-legacy:
+
+Upgrade from the Legacy API
+---------------------------
+
+In order to upgrade from the legacy API, packaged as the ``driver-legacy``
+JAR, to the current API, packaged as the ``driver-sync`` JAR, you must
+update the following references to classes and methods:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+   :class: compatibility-large
+
+   * - Legacy
+     - Current
+
+   * - `MongoClientOptions <{+api+}/apidocs/mongodb-driver-legacy/com/mongodb/MongoClientOptions.html>`__
+     - `MongoClientSettings <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html>`__
+
+   * - `new MongoClient() <{+api+}/apidocs/mongodb-driver-legacy/com/mongodb/MongoClient.html>`__
+     - `MongoClients.create() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClients.html#create()>`__
+
+   * - `MongoClient.getDB() <{+api+}/apidocs/mongodb-driver-legacy/com/mongodb/MongoClient.html#getDB(java.lang.String)>`__
+     - `MongoClient.getDatabase() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html#getDatabase(java.lang.String)>`__
+
+   * - `DB <{+api+}/apidocs/mongodb-driver-legacy/com/mongodb/DB.html>`__
+     - `MongoDatabase <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html>`__
+
+   * - `DBCollection <{+api+}/apidocs/mongodb-driver-legacy/com/mongodb/DBCollection.html>`__
+     - `MongoCollection<> <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html>`__
+
+   * - `DBCursor <{+api+}/apidocs/mongodb-driver-legacy/com/mongodb/DBCursor.html>`__
```

Comment:
This is incomplete.  In the legacy API, DBCursor takes on the role of both MongoCursor and FindIterable, whereas in the new API it's split. Perhaps we could give more detailed examples showing the difference.

Reply:
Ok, I can add a note about that and ticket the work to create examples for each. Does that work?

Reply:
Created: https://jira.mongodb.org/browse/DOCSP-21349

Reply:
Sure